x-pack/plugins/observability/public/components/shared/field_value_suggestions/field_value_selection.tsx

```diff
@@ -95,7 +95,8 @@ export function FieldValueSelection({
 
   useEffect(() => {
     setOptions(formatOptions(values, selectedValue, excludedValue, showCount));
-  }, [values, selectedValue, showCount, excludedValue]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(values), JSON.stringify(selectedValue), showCount, excludedValue]);
```

Comment:
Doesn't this impact performance since it uses JSON.stringify on every render?
Can't we use `useMemo` in the parent component instead?

Reply:
i think in this specific case. it won't have any impact. Since it's only meant to be triggered on new fetch of the data.

Reply:
if the new fetch of the data remains same, we don't want existing component to re-render since data selection should remain same.

Reply:
>i think in this specific case. it won't have any impact. Since it's only meant to be triggered on new fetch of the data.
You mean `FieldValueSelection` only re-renders when there is new data? Where is this logic?

> Using useMemo won't help in parent component.
Why is that?

Reply:
this is being used on auto-refresh in synthetics app. so when that happens. the new data remains same , but it will trigger useMemo re-render. this is why we are doing this deep comparison. Json.stringify impact is minimal here. since data isn't big.